src/testdir/test_xxd.vim

```diff
@@ -411,6 +411,17 @@ func Test_xxd_max_cols()
   endfor
 endfunc
 
+
+" Try to trigger a buffer overflow (#14738)
+func Test_xxd_buffer_overflow()
+  new
+  let input = repeat('A', 256)
+  let expected = ' -9223372036854775808: ' . repeat("\e[1;32m41\e[0m ", 256) . ' ' . repeat("\e[1;32mA\e[0m", 256)
```

Comment:
I don't know why the expected result starts with a space (it does not seem be printed by xxd). I also think the offset shouldn't be negative but it is what it is.